packages/expo-sqlite/src/SQLiteDatabase.ts

```diff
@@ -407,20 +407,63 @@ export class SQLiteDatabase {
   //#endregion
 }
 
+/**
+ * The default directory for SQLite databases.
+ */
+export const defaultDatabaseDirectory = ExpoSQLite.defaultDatabaseDirectory;
+
+/**
+ * Resolves the database directory from the given directory or the default directory.
+ *
+ * @hidden
+ */
+export function resolveDbDirectory(directory: string | undefined): string {
+  const resolvedDirectory = directory ?? defaultDatabaseDirectory;
+  if (resolvedDirectory === null) {
+    throw new Error('Both provided directory and defaultDatabaseDirectory are null.');
+  }
+  return resolvedDirectory;
+}
+
+/**
+ * Creates a normalized database path by combining the directory and database name.
+ *
+ * Ensures the directory does not end with a trailing slash and the database name
+ * does not start with a leading slash, preventing redundant slashes in the final path.
+ *
+ * @hidden
+ */
+export function createDatabasePath(databaseName: string, directory: string): string {
+  if (databaseName === ':memory:') return databaseName;
+
+  function removeTrailingSlash(path: string): string {
+    return path.replace(/\/*$/, '');
+  }
+  function removeLeadingSlash(path: string): string {
+    return path.replace(/^\/+/, '');
+  }
+
+  return `${removeTrailingSlash(directory)}/${removeLeadingSlash(databaseName)}`;
+}
+
 /**
  * Open a database.
  *
  * @param databaseName The name of the database file to open.
  * @param options Open options.
+ * @param directory The directory where the database file is located. The default value is `defaultDatabaseDirectory`.
  */
 export async function openDatabaseAsync(
   databaseName: string,
-  options?: SQLiteOpenOptions
+  options?: SQLiteOpenOptions,
+  directory?: string
 ): Promise<SQLiteDatabase> {
   const openOptions = options ?? {};
-  const nativeDatabase = new ExpoSQLite.NativeDatabase(databaseName, openOptions);
+  const databasePath = createDatabasePath(databaseName, resolveDbDirectory(directory));
```

Comment:
there's too many occurrences of  `createDatabasePath(databaseName, resolveDbDirectory(directory));`. maybe we can move this to`pathUtils.ts` and have `function createDatabasePath(databaseName: string, directory?: string` there. then we can also do some js unit test there